src/diff_walker.rs

```diff
@@ -27,21 +27,43 @@ impl DiffWalker {
         if let (Some(lhs_any_of), Some(rhs_any_of)) =
             (&mut lhs.subschemas().any_of, &mut rhs.subschemas().any_of)
         {
-            lhs_any_of.sort_by_cached_key(|x| format!("{x:?}"));
-            rhs_any_of.sort_by_cached_key(|x| format!("{x:?}"));
+            match (lhs_any_of.len(), rhs_any_of.len()) {
+                (l, r) if l <= r => {
+                    lhs_any_of.append(&mut vec![Schema::Bool(false); r - l]);
+                }
+                (l, r) => {
+                    rhs_any_of.append(&mut vec![Schema::Bool(false); l - r]);
+                }
+            }
 
-            for (i, (lhs_inner, rhs_inner)) in
-                lhs_any_of.iter_mut().zip(rhs_any_of.iter_mut()).enumerate()
-            {
+            let len = lhs_any_of.len();
+            let mut mat = pathfinding::matrix::Matrix::new(len, len, 0i64);
+            for (i, l) in lhs_any_of.iter_mut().enumerate() {
+                for (j, r) in rhs_any_of.iter_mut().enumerate() {
+                    let mut walker = DiffWalker {
+                        changes: vec![],
+                        lhs_root: self.lhs_root.clone(),
+                        rhs_root: self.rhs_root.clone(),
+                    };
+                    walker.diff(
+                        "",
+                        &mut l.clone().into_object(),
+                        &mut r.clone().into_object(),
+                    )?;
+                    mat[(i, j)] = walker.changes.len() as i64;
```

Comment:
It would be more convenient for `Diffwalker::diff` to return `Vec<Change>` instead of mutating internal `changes`. By doing so we can reuse `Diffwalker` and avoid cloning `RootSchema` here